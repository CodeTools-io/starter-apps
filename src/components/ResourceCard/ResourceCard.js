import React from 'react';

import { Box, Card, CardBody, Heading, Image, Paragraph } from 'grommet';

export default function ResourceCard({
  children,
  title = '',
  description = '',
  image = '',
  imageTitle = '',
  ...props
}) {
  return (
    <Box pad="medium" align="start" {...props}>
      <Card elevation="none" width="medium">
        <CardBody height="small">
          <Image fit="cover" src={image} a11yTitle={imageTitle} />
        </CardBody>
        <Box pad={{ horizontal: 'medium' }} responsive={false}>
          <Heading level="3" margin={{ vertical: 'medium' }}>
            {title}
          </Heading>
          <Paragraph margin={{ top: 'none' }}>{description}</Paragraph>
        </Box>
      </Card>
    </Box>
  );
}
