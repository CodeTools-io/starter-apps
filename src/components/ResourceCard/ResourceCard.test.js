import React from 'react'
import ResourceCard from './ResourceCard'

describe('ResourceCard', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <ResourceCard>
        <p>Content for the ResourceCard</p>
      </ResourceCard>
    )

    expect(container).toMatchSnapshot()
  })
})

