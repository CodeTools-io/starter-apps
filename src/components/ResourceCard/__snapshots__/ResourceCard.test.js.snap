// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCard can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 bEJYwZ"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 cFuWap"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 dPcePz"
      >
        <img
          aria-label=""
          class="StyledImage-ey4zx9-0 dQECSV"
          src=""
        />
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xnWsV"
      >
        <h3
          class="StyledHeading-sc-1rdh4aw-0 craNUF"
        />
        <p
          class="StyledParagraph-tbetod-0 byDKns"
        />
      </div>
    </div>
  </div>
</div>
`;
